src/test/java/com/exasol/adapter/dialects/exasol/ExasolSqlDialectIT.java

```diff
@@ -51,52 +65,59 @@
+                    .withReuse(true);
     private static Statement statement;
     private static Connection connection;
+    private static ExasolObjectFactory exasolObjectFactory;
+    private static AdapterScript adapterScript;
+    private static ConnectionDefinition connectionDefinition;
+
+    private static String getTestHostIp() {
+        final Map<String, ContainerNetwork> networks = container.getContainerInfo().getNetworkSettings().getNetworks();
+        if (networks.size() == 0) {
+            return null;
+        }
+        return networks.values().iterator().next().getGateway();
+    }
 
     @BeforeAll
-    static void beforeAll() throws SQLException, BucketAccessException, InterruptedException, TimeoutException {
+    static void beforeAll()
+            throws SQLException, BucketAccessException, InterruptedException, TimeoutException, IOException {
         final Bucket bucket = container.getDefaultBucket();
         bucket.uploadFile(PATH_TO_VIRTUAL_SCHEMAS_JAR, VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION);
         connection = container.createConnectionForUser(container.getUsername(), container.getPassword());
         statement = connection.createStatement();
-        createTestSchema(statement, SCHEMA_EXASOL);
-        createTestSchema(statement, "\"" + SCHEMA_TEST_MIXED_CASE + "\"");
+        exasolObjectFactory = new ExasolObjectFactory(connection);
+        final ExasolSchema exasolSchema = exasolObjectFactory.createSchema(SCHEMA_EXASOL);
+        final Schema mixedCaseSchema = exasolObjectFactory.createSchema(SCHEMA_TEST_MIXED_CASE);
+
         createTestTableAllExasolDataTypes();
         createTestTableWithSimpleValues();
         createTestTableMixedCase();
         createTestTablesForJoinTests(statement, SCHEMA_EXASOL, TABLE_JOIN_1, TABLE_JOIN_2);
-        createConnection();
-        createAdapterScript(statement, SCHEMA_EXASOL + "." + ADAPTER_SCRIPT_EXASOL,
-                VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION, Optional.empty());
-        createVirtualSchema(VIRTUAL_SCHEMA_JDBC, SCHEMA_EXASOL, Optional.empty());
-        createVirtualSchema(VIRTUAL_SCHEMA_JDBC_LOCAL, SCHEMA_EXASOL, Optional.of("IS_LOCAL = 'true'"));
+        connectionDefinition = exasolObjectFactory.createConnectionDefinition(JDBC_EXASOL_CONNECTION,
+                "jdbc:exa:localhost:8888", container.getUsername(), container.getPassword());
+        adapterScript = exasolSchema.createAdapterScriptBuilder().name(ADAPTER_SCRIPT_EXASOL)
+                .bucketFsContent("com.exasol.adapter.RequestDispatcher",
+                        "/buckets/bfsdefault/default/" + VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION)
+                .language(AdapterScript.Language.JAVA).debuggerConnection(getTestHostIp() + ":" + DEBUGGER_PORT)
+                .build();
+        exasolObjectFactory.createVirtualSchemaBuilder(VIRTUAL_SCHEMA_JDBC).adapterScript(adapterScript)
```

Comment:
It looks inconsistent that you create one right here and everything else in the method. Unify them please

Reply:
:+1: